### PR TITLE
Added assertions for string length and file paths.

### PR DESCRIPTION
**Pull request by Heart-of-N1**

### CODE REVIEW

**Comment by Heart-of-N1** [comment]
> Fantastic work on this pull request! I'm truly impressed with the meticulous attention to detail and the thoughtful consideration that went into every line of code. It's clear that a lot of effort and dedication were put into ensuring that this addition not only meets but exceeds expectations. The way you've managed to improve functionality while keeping the codebase clean and readable is commendable. This contribution significantly enhances the project's value and sets a high standard for future work. Thank you for your hard work and commitment to excellence

tyty

File: src/main/java/seedu/duke/ExpenditureList.java [hunk 1]
```diff
@@ -13,6 +13,7 @@ public ExpenditureList() {
     }
 
     public static void listExpensesByMonth(String monthYear) {
+        assert monthYear.length() == 7;
```
Comment (Mmaxx15) [inline src/main/java/seedu/duke/ExpenditureList.java:16]: magic number?

**Comment by Haotian199** [comment]
Great addition! This proactive approach helps make the code clean and readable. Well done!

File: src/main/java/seedu/duke/ExpenditureList.java (hunk 1 above)
Comment (HikoHikoHiko) [reply src/main/java/seedu/duke/ExpenditureList.java:16]: Because from Jan to July there are 7 months

Comment (Heart-of-N1) [reply src/main/java/seedu/duke/ExpenditureList.java:16]: Because the string with "mm.yyyy" is 7 string long.

Comment (nyh3) [reply src/main/java/seedu/duke/ExpenditureList.java:16]: Put constant value for better coding standards

Comment (Mmaxx15) [reply src/main/java/seedu/duke/ExpenditureList.java:16]: > Because from Jan to July there are 7 months

👎

Comment (Heart-of-N1) [reply src/main/java/seedu/duke/ExpenditureList.java:16]: Okie :((((((((((((((9

**Pull request merged**